### PR TITLE
#5 Allow setting a proxy

### LINKED ISSUE #5: Allow to configure Guzzle options / use a proxy to ensure static IP address
The Bunq API expects requests to be initiated from a fixed / whitelisted IP address, which is difficult in a "cloud" environment, with ever changing IP addresses. 

A proxy can circumvent this problem and ensure a fixed IP address is used. However the current [initialisation of the Guzzle client](https://github.com/bunq/sdk_php/blob/93ab9b472b389ae02fffab1de33df81c959794e3/lib/Http/ApiClient.php#L154-L166) does not allow to set [the `proxy` option](docs.guzzlephp.org/en/stable/request-options.html#proxy).  

Also see this issue which discusses a similar situation: https://github.com/link0/bunq/issues/27

### PR DESCRIPTION
**Pull request by qurben**
Save the proxy in the ApiContext. Allow an api context to not contain a proxy.

Fixes #5

### CODE REVIEW

**Comment by dnl-blkv** [comment]
@qurben Have you tested it somehow? :)

**Comment by qurben** [comment]
@dnl-blkv I tested it with a socks proxy through ssh to a remote server.

File: README.md [hunk 1]
```diff
@@ -80,6 +80,20 @@ $apiContext = ApiContext::restore($fileName);
 **Tip:** both saving and restoring the context can be done without any arguments. In this case the context will be saved
 to/restored from the `bunq.conf` file in the same folder with your script.
 
+##### Proxy
+
+You can use a proxy with the bunq PHP SDK. This option must be a string. This proxy will be used for all requests done with
+the SDK. You will be prompted to provide a proxy URL when using the interactive installation script.
```
Comment (dnl-blkv) [inline README.md]: May be smth like `This proxy will be used for all the requests done with the context for which it was specified` ?

File: README.md [hunk 2]
```diff
@@ -80,6 +80,20 @@ $apiContext = ApiContext::restore($fileName);
 **Tip:** both saving and restoring the context can be done without any arguments. In this case the context will be saved
 to/restored from the `bunq.conf` file in the same folder with your script.
 
+##### Proxy
+
```
Comment (dnl-blkv) [inline README.md]: Do we really need this newline? :P

File: README.md [hunk 3]
```diff
@@ -80,6 +80,20 @@ $apiContext = ApiContext::restore($fileName);
...
+You can use a proxy with the bunq PHP SDK. This option must be a string. This proxy will be used for all requests done with
+the SDK. You will be prompted to provide a proxy URL when using the interactive installation script.
+
+```php
+$proxyUrl = 'socks5://localhost:1080'; // The proxy for all requests, leave empty to disable
```
Comment (dnl-blkv) [inline README.md]: Leave empty? So, empty string disables it?

File: src/Context/ApiContext.php [hunk 4]
```diff
@@ -331,6 +346,14 @@ public function getEnvironmentType()
     }
 
     /**
+     * @return string
```
Comment (dnl-blkv) [inline src/Context/ApiContext.php]: |null

File: src/Context/ApiContext.php [hunk 5]
```diff
@@ -331,6 +346,14 @@ public function getEnvironmentType()
...
+     */
+    public function getProxyUrlOrNull()
```
Comment (dnl-blkv) [inline src/Context/ApiContext.php]: We typically do not do OrNull in the getters in the SDK code. The reason is that the option of nullability is already stated in the doc header, and adding it into method name is not a normal practice among the PHP community.

**Pull request merged**